### PR TITLE
Us admin partners

### PR DESCRIPTION
**Pull request by CeliaBoiss**

### CODE REVIEW

**Review by BoulMaker — DISMISSED** [review]

**Review by Quentin-BLANC — DISMISSED** [review]

**Review by BoulMaker — DISMISSED** [review]

**Review by Quentin-BLANC — DISMISSED** [review]

**Review by BoulMaker — DISMISSED** [review]

**Review by Quentin-BLANC — DISMISSED** [review]

File: migrations/Version20210111105323.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        return '';
+    }
+
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('CREATE TABLE partner (id INT AUTO_INCREMENT NOT NULL, name VARCHAR(255) NOT NULL, url VARCHAR(255) NOT NULL, logo VARCHAR(255) NOT NULL, type VARCHAR(255) NOT NULL, PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('CREATE TABLE user (id INT AUTO_INCREMENT NOT NULL, email VARCHAR(180) NOT NULL, roles JSON NOT NULL, password VARCHAR(255) NOT NULL, UNIQUE INDEX UNIQ_8D93D649E7927C74 (email), PRIMARY KEY(id)) DEFAULT CHARACTER SET utf8mb4 COLLATE `utf8mb4_unicode_ci` ENGINE = InnoDB');
+        $this->addSql('DROP TABLE housing');
```
Comment (RomainClair) [inline migrations/Version20210111105323.php]: Your migrations are inconsistent. No previous accepted migration create a housing table and you drop it there.
In consequence, applying the repository migrations fails

File: src/Controller/HomeController.php [hunk 2]
```diff
@@ -35,6 +73,28 @@ public function index(Request $request, MailerInterface $mailer): Response
...
+
+    /**
+     * @Route("/{id}", name="partner_delete", methods={"DELETE"})
+     */
+    public function deletePartner(Request $request, Partner $partner, FileManager $fileManager): Response
+    {
+        if ($this->isCsrfTokenValid('delete' . $partner->getId(), $request->request->get('_token'))) {
+            $fileManager->deleteFile($partner->getLogo(), $this->getParameter('partners_directory'));
+
+            $entityManager = $this->getDoctrine()->getManager();
```
Comment (RomainClair) [inline src/Controller/HomeController.php]: Use dependecy injection

File: src/Services/FileManager.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+use Symfony\Component\HttpFoundation\File\Exception\FileException;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+use Symfony\Component\Filesystem\Filesystem;
+
+class FileManager
+{
+
+    /**
+     * @return array<string>
+     */
```
Comment (RomainClair) [inline src/Services/FileManager.php]: You really should have comments explaining this method purpose

File: src/Services/FileManager.php [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        $filesystem = new Filesystem();
+        $path = $folder . $fileName;
+        $filesystem->remove($path);
+    }
+
+    public function getNewFileName(string $oldName, string $delimiter = '-'): ?string
+    {
+        $newName = iconv('UTF-8', 'ASCII//TRANSLIT', $oldName);
+
+        if ($newName != null) {
```
Comment (RomainClair) [inline src/Services/FileManager.php:53]: You should have strict comparison by default

**Review by RomainClair — CHANGES_REQUESTED** [review]
You should have only static content in the repository, so the dynamic images upload directory should be in the .gitignore

File: src/Services/FileManager.php (hunk 4 above)
Comment (CeliaBoiss) [reply src/Services/FileManager.php:53]: Stan bother me again when I put !==, he say the same error message as before :   54     Parameter #3 $subject of function preg_replace expects array|string,  
         string|false given. 
So I let != and he's fine with it

**Review by Quentin-BLANC — APPROVED** [review]

**Review by BoulMaker — APPROVED** [review]

**Review by thocle86 — APPROVED** [review]

**Review by RomainClair — APPROVED** [review]

**Pull request merged**